tests/net/hostname/src/main.c

```diff
@@ -381,6 +381,13 @@ ZTEST(net_hostname, test_hostname_set)
 		zassert_equal(ret, -EALREADY,
 			      "Could set hostname postfix (%d)", ret);
 	}
+
+	if (IS_ENABLED(CONFIG_NET_HOSTNAME_DYNAMIC)) {
+		int ret;
+
+		ret = net_hostname_set("foobar", sizeof("foobar") - 1);
+		zassert_equal(ret, 0, "Could not set hostname (%d)", ret);
+	}
```

Comment:
We should also verify that the set hostname is really set correctly, so need to call `net_hostname_get()` and verify that it is really the one we set earlier.
The getter could be called after this if() block so that it could be shared with the unique if that is easily doable.